### PR TITLE
[1.20.6] Fix Biome Decoration Crash.

### PR DESCRIPTION
**Pull request by RealMangorage**
Test Mod provided for testing purposes inside of test folder.

This will fix the crash

### CODE REVIEW

**Comment by Jonathing** [comment]
Duplicate of #9975. If for whatever reason that one cannot be merged, this one can be considered in its place.

File: patches/minecraft/net/minecraft/world/level/chunk/ChunkGenerator.java.patch [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ 
+     public ChunkGenerator(BiomeSource p_256133_) {
+@@ -91,9 +_,14 @@
+     public ChunkGenerator(BiomeSource p_255838_, Function<Holder<Biome>, BiomeGenerationSettings> p_256216_) {
+         this.biomeSource = p_255838_;
+         this.generationSettingsGetter = p_256216_;
+-        this.featuresPerStep = Suppliers.memoize(
+-            () -> FeatureSorter.buildFeaturesPerStep(List.copyOf(p_255838_.possibleBiomes()), p_223216_ -> p_256216_.apply(p_223216_).features(), true)
++        this.featuresPerStep = net.minecraftforge.common.util.ClearableLazy.concurrentOf(
++                () -> FeatureSorter.buildFeaturesPerStep(List.copyOf(p_255838_.possibleBiomes()), p_223216_ -> p_256216_.apply(p_223216_).features(), true)
```
Comment (LexManos) [inline patches/minecraft/net/minecraft/world/level/chunk/ChunkGenerator.java.patch]: Why are you tabbing this in?

File: src/main/java/net/minecraftforge/common/util/ClearableLazy.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    {
+        return new ClearableLazy.Fast<T>(supplier);
+    }
+
+    /**
+     * Constructs a thread-safe lazy-initialized object
+     * @param supplier The supplier for the value, to be called the first time the value is needed.
+     */
+    static <T> ClearableLazy<T> concurrentOf(@NotNull Supplier<T> supplier)
+    {
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/common/util/ClearableLazy.java]: all these braces should be on the new line as per the new standard

File: src/test/java/net/minecraftforge/debug/gameplay/biome/DataGen.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+@Mod.EventBusSubscriber(modid = BiomeTestMod.MOD_ID, bus = Mod.EventBusSubscriber.Bus.MOD)
+public class DataGen {
+
+    @SubscribeEvent
+    public static void gatherdata(GatherDataEvent event) {
+        DataGenerator generator = event.getGenerator();
```
Comment (LexManos) [inline src/test/java/net/minecraftforge/debug/gameplay/biome/DataGen.java]: tis doesnt need to exist?

**Pull request merged**